### PR TITLE
Expose Buffer2D and ParallelHelper internals

### PR DESCRIPTION
**Pull request by antonfirsov**
### Prerequisites

- [x] I have written a descriptive pull-request title
- [x] I have verified that there are no overlapping [pull-requests](https://github.com/SixLabors/ImageSharp/pulls) open
- [x] I have verified that I am following matches the existing coding patterns and practice as demonstrated in the repository. These follow strict Stylecop rules :cop:.
- [x] I have provided test coverage for my change (where applicable)

### Description
Sister PR of #1028. Fix more points of #967.
API-s being exposed:
- `Buffer2D<T>` and `Buffer2DExtensions`: minimum API surface needed by Drawing
- `ParallelHelper`:  and `ParallelExecutionSettings`: 
  - Everything but `IterateRowsWithTempBuffer`
  - Both types moved to namespace `SixLabors.ImageSharp.Advanced.ParallelUtils`

Additional:
Fixed error cases for `Configuration.MaxDegreeOfParallelism` and `ParallelExecutionSettings.MaxDegreeOfParallelism`.

### CODE REVIEW

File: src/ImageSharp/Advanced/ParallelUtils/ParallelHelper.cs [hunk 1]
```diff
@@ -81,7 +82,7 @@ public static void IterateRows(Rectangle rectangle, Configuration configuration,
         /// Iterate through the rows of a rectangle in optimized batches defined by <see cref="RowInterval"/>-s
         /// instantiating a temporary buffer for each <paramref name="body"/> invocation.
         /// </summary>
-        public static void IterateRowsWithTempBuffer<T>(
+        internal static void IterateRowsWithTempBuffer<T>(
```
[the remark this answers is not in the document]
Comment (antonfirsov) [reply src/ImageSharp/Advanced/ParallelUtils/ParallelHelper.cs]: Yes. I'm not fully happy with this API yet, so I'd keep it internal for now.

File: src/ImageSharp/Memory/Buffer2D{T}.cs [hunk 2]
```diff
@@ -12,8 +12,12 @@ namespace SixLabors.ImageSharp.Memory
     /// Represents a buffer of value type objects
     /// interpreted as a 2D region of <see cref="Width"/> x <see cref="Height"/> elements.
     /// </summary>
+    /// <remarks>
+    /// Before RC1, this class might be target of API changes, use it on your own risk!
+    /// </remarks>
     /// <typeparam name="T">The value type.</typeparam>
-    internal sealed class Buffer2D<T> : IDisposable
+    // TODO: Consider moving this type to the SixLabors.Memory namespace (SixLabors.Core).
```
[the remark this answers is not in the document]
Comment (antonfirsov) [reply src/ImageSharp/Memory/Buffer2D{T}.cs]: Not 100% sure yet, depends on future investigations.

**Comment by antonfirsov** [comment]
@JimBobSquarePants wow that was quick! Does it mean that you are happy with my namespace choice `SixLabors.ImageSharp.Advanced.ParallelUtils`? I wasn't 100% sure about that.

**Comment by JimBobSquarePants** [comment]
Yeah, happy with that, certainly advanced and Utils is a common suffix.

**Pull request merged**